lib/bson/open_struct.rb

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2009-2014 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module BSON
+  # Injects behaviour for encoding OpenStruct objects using hashes
+  # to # raw bytes as specified by the BSON spec.
+  #
+  # @see http://bsonspec.org/#/specification
+  #
+  # @since 2.0.0
+  module OpenStruct
+
+    # Get the OpenStruct as encoded BSON.
+    #
+    # @example Get the hash as encoded BSON.
+    #   OpenStruct.new({ "field" => "value" }).to_bson
+    #
+    # @return [ String ] The encoded string.
+    #
+    # @see http://bsonspec.org/#/specification
+    #
+    # @since 2.0.0
+    def to_bson(buffer = ByteBuffer.new, validating_keys = Config.validating_keys?)
+      if RUBY_VERSION < "2.0.0"
```

Comment:
Have we discussed dropping 1.9.x support? EOL for 1.9.3 was a year and a half ago. (https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/) If we can drop 1.9.3 support we can remove the marshal dump here...